client/landing/gutenboarding/components/signup-form/index.tsx

```diff
@@ -167,13 +167,25 @@ const SignupForm = ( { onRequestClose }: Props ) => {
 	}
 
 	const langFragment = lang ? `/${ lang }` : '';
-	const loginRedirectUrl = encodeURIComponent(
-		isAnchorFmSignup
-			? `${ window.location.origin }/new${ makePath(
-					Step.IntentGathering
-			  ) }?new&anchor_podcast=${ anchorFmPodcastId }` // Fix in #48389
-			: `${ window.location.origin }/new${ makePath( Step.CreateSite ) }?new`
-	);
+
+	let loginRedirectUrl = window.location.origin + '/new';
+	if ( isAnchorFmSignup ) {
+		loginRedirectUrl += `${ makePath( Step.IntentGathering ) }?new`;
+		const queryParts = {
+			anchor_podcast: anchorFmPodcastId,
+			anchor_episode: anchorFmEpisodeId,
+			spotify_show_url: anchorFmSpotifyShowUrl,
+		};
+		for ( const [ k, v ] of Object.entries( queryParts ) ) {
```

Comment:
I was thinking this could be more explicit, but k for key and v and value is awesome 👍